workspace/2019 DeepSpace/src/main/java/frc/robot/Lifter.java

```diff
@@ -42,75 +42,92 @@ public void Lower() {
         m_lifter.set(-kMotorPowerLevel);
     }
     
+    public void Stop() {
+        m_lifter.set(0);
+    }
+    
     public void GoToBottom() {
         if (m_hight.atBottom()) {
             m_lifter.set(0);
             loc = Location.bottom;
-        } else { m_lifter.set(-kMotorPowerLevel); }
+        } else {
+            m_lifter.set(-kMotorPowerLevel);
+        }
     }
+    
     public void GoToTop() {
         if (m_hight.atTop()) {
             m_lifter.set(0);
             loc = Location.top;
-        } else { m_lifter.set(kMotorPowerLevel); }
+        } else {
+            m_lifter.set(kMotorPowerLevel);
+        }
     }
 
+    // All these go to level functions would be much easier in python.
     public void GoToFirstBallLevel() {
         if (m_hight.atFirstBallLevel()) {
             m_lifter.set(0);
             loc = Location.ballFirstLevel;
-        }else if (m_hight.atBottom()) {
-            m_lifter.set(kMotorPowerLevel); 
-        }else if (m_hight.atSecondBallLevel() || m_hight.atThirdBallLevel()){
-            m_lifter.set(-kMotorPowerLevel); }  
+        } else if (m_hight.getDistance() < m_hight.ballFirstLevelHight) {
+            m_lifter.set(kMotorPowerLevel);
+        } else if (m_hight.getDistance() > m_hight.ballFirstLevelHight) {
+            m_lifter.set(-kMotorPowerLevel);
+        }
     }
+    
     public void GoToSecondBallLevel() {
         if (m_hight.atSecondBallLevel()) {
             m_lifter.set(0);
             loc = Location.ballSecondLevel;
-        }else if (m_hight.atBottom() || m_hight.atFirstBallLevel()) {
-            m_lifter.set(kMotorPowerLevel); 
-        }else if (m_hight.atThirdBallLevel() || m_hight.atTop()){
-            m_lifter.set(-kMotorPowerLevel); }  
+        } else if (m_hight.getDistance() < m_hight.ballSecondLevelHight) {
```

Comment:
Rather than exposing these private values, this could instead be:
```
if (m_hight.atSecondBallLevel()) {
    m_lifter.set(0);
    loc = Location.ballSecondLevel;
} else {
    m_lifter.goToSecondBallLevel();
}
```

Reply:
That becomes a recursive function. `m_lifter.GoToSecondBallLevel()` is what is being defined. It requires the lifter height locations to know whether to move up or down.